test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -3039,7 +3039,8 @@ BOOST_AUTO_TEST_CASE(gaslimit)
 		}
 	)";
 	compileAndRun(sourceCode);
-	ABI_CHECK(callContractFunction("f()"), encodeArgs(gasLimit()));
```

Comment:
This caused the ``gasLimit`` test failure.
Since we call this in one line the order of execution is undefined and ``gasLimit()`` was sometimes called before the actual transaction.
Since in older version of ``aleth`` the ``gasLimit`` was constant, this hasn't been a problem before, but now we need to make sure that ``gasLimit()`` is in fact called *after* the transaction.